01-arch/02-policy.md

```diff
@@ -1 +1,108 @@
+the set of nodes which it specifies.
+The exact meaning should be obvious from the context.
+
+#### Simple filters
+Simple filter can compare single node attribute with some value.
+It has 3 fields:
+1. `Key` -- name of node attirbute. 
+2. `Value` -- value to compare attibute with.
+3. `Op` -- operation to used for comparison.
+
+To simplify understanding simple filters will be specified as `Key Op Value`.
+
+For example:
+1. `Country = USA` means "use nodes for which `Country` attribute
+   is equal to `USA`", i.e. nodes located in USA.
+2. `Price < 100` means "use nodes which specify 100 price per unit of storage". 
+   Price units are agreed upon by each NeoFS network
+   and are beyond the scope of this document.
+
+#### Operations
+Currently only 8 operations are allowed, 2 of which are used for creating
+compound filters from other ones.
+1. `EQ`/`NE` check if attribute is equal/not equal to filter's value.
+2. `GT`/`GE`/`LT`/`LE` check if numerical attribute
+   is greater-than/greater-or-equal/less-than/less-or-equal than
+   filter's value.
+3. `OR` checks if node satisfies at least one of filters provided as arguments.
+4. `AND` checks if node satisfies all filters provided as arguments.
+
+#### Compound filters
+Compound filter can combine simple filters to specify arbitrarily complex
+conditions. Consider example from the previous section: we may write
+`Country = USA AND Price < 100` to filter nodes which are located in USA
+and have price less than 100.
+#### Filters with name
+If filters are used in selectors (or other filters) they should have name.
+Consider filter `GoodCountry`: `Country = Spain OR Country = Canada`.
+If we need nodes from these countries but want to vary maximum price
+depending on the storage type they have we may write this filter:
+```
+GoodCountry AND StorageType = SSD AND Price < 100
+OR
+GoodCountry AND StorageType = HDD AND Price < 10
+```
+### Selectors
+Selector is a mechanism to specify which of the nodes previously
+filtered will be included in the container. It has 5 fields:
+1. `Name` -- name that can be referred to.
+2. `Attribute` -- name of the attribute for grouping nodes.
+When it is set, nodes are grouped in buckets based on `Attribute` value.
+It can be omitted to create buckets "randomly".
+3. `Count` -- how many nodes to choose bucket.
+4. `Clause` specifies how `Count` is interpreted:
+   - `SAME` -- choose nodes from the same bucket
+   - `DISTINCT` -- choose nodes from distinct buckets.
+5. `Filter` name of the filter to choose nodes from.
+   If it is omitted or is `*`, all nodes from netmap are used. 
+
+Selector can return different set of nodes for every epoch,
+however they are always the same on each storage node having the same netmap.
+The degree to which this set of nodes is changed also depends on how strict
+the filter is. For example, if we specify a few nodes precisely this set
+will always be the same. However if all of these nodes go down data can be lost.
```

Comment:
"if we specify a few nodes precisely this set
will always be the same" - do you mean that you have to specify nodes precisely enough? "Precisely" alone doesn't work for me in this context.

Reply:
Rewrote with _For example, if we select a few nodes based on very specific
attributes, this set will always be the same._.
The meaning is that the stricter storage policy is, the less selected nodes change between epochs.